File: AddOns/Share/Categories/StopVisits+Shareable.m
//
//  StopVisits+Shareable.m
//  Pods
//
//  Created by Adrian Schoenig on 24/06/2016.
//
//

#import "StopVisits+Shareable.h"

#import "TKShareHelper.h"
#import "StopLocation.h"

@implementation StopVisits (Shareable)

- (NSURL *)shareURL
{
  return [TKShareHelper meetURLForCoordinate:[self.stop coordinate] atTime:self.time];
  
  // Once the web app supports it: https://redmine.buzzhives.com/issues/2200
  //  return [ShareHelper serviceURLForServiceID:self.service.code
  //                                  atStopCode:self.stop.stopCode
  //                               inRegionNamed:self.stop.region.name];
}

@end

File: AddOns/Share/Categories/StopVisits+Shareable.swift
//
//  StopVisits+Shareable.swift
//  TripKit
//
//  Created by Adrian Schoenig on 30/6/17.
//  Copyright © 2017 SkedGo. All rights reserved.
//

import Foundation

#if TK_NO_MODULE
#else
  import TripKit
#endif

extension StopVisits: TKURLShareable {
  
  public var shareURL: URL? {
    guard let regionName = stop.region?.name else { return nil }
    return TKShareHelper.createServiceURL(serviceID: service.code, atStopCode: stop.stopCode, inRegionNamed: regionName)
  }
  
}
